### PR TITLE
Forbidden project names

### PR DESCRIPTION
**Pull request by Aryojaam**
# Why
This pull request resolves the following issue:
https://github.com/expo/expo-cli/issues/3979

If project names such as `react-native` or `expo` is fed to the command  `expo init <project_name>`, it leads to the project being correctly initialized however crashing upon starting up. These edge cases should be caught and handled. 

# How
There is an array of blacklisted project names. Upon initializing the project, it is checked if the project name is included in this array. If so, the user is stopped from creating the project with that name.  

Here is how the result looks like:
Screenshot from 2021-11-21 15-51-02
# Test Plan
For testing using the command line simply run `node ./packages/expo-cli/bin/expo.js init react` and the error should pop up. Make sure the project is built and running in the background using `yarn build` and `yarn start` respectively.  

There is also an E2E test case for this fix that checks if the command has failed by checking if `stdout` is empty in comparison to a passing state in which the `stdout` is `Your project is ready!`. However currently a warning gets thrown on executing the test. The warning is present in the following screenshot:
expotest

### CODE REVIEW

**Comment by axeldelafosse** [comment]
Would be cool to avoid the term "blacklist" :)

**Comment by Aryojaam** [comment]
Sure, I will refactor that :)

File: packages/expo-cli/src/commands/initAsync.ts [hunk 1]
```diff
@@ -64,6 +64,10 @@ function assertValidName(folderName: string) {
       `Cannot create an app named ${chalk.red(`"${folderName}"`)}. ${validation}`
     );
   }
+  const isFolderNameForbidden = CreateApp.isFolderNameForbidden(folderName);
+  if (isFolderNameForbidden) {
+    throw new CommandError(`Cannot create an app named ${chalk.red(`"${folderName}"`)}.`);
```
Comment (EvanBacon) [inline packages/expo-cli/src/commands/initAsync.ts:69]:
```suggestion
    throw new CommandError(`Cannot create an app named ${chalk.red(`"${folderName}"`)} because it would conflict with a dependency of the same name.`);
```

**Review by EvanBacon — APPROVED** [review]
lgtm

**Comment by Aryojaam** [comment]
Cool! Thanks! I have committed your suggestion :)
